skyline-vscode/react-ui/src/utils/utils.js

```diff
@@ -139,82 +133,123 @@ export function getTraceByLevel(tree) {
       forward_ms: 0,
       backward_ms: 0,
       size_bytes: 0,
-      total_time: total_time - tracked_total_time
+      total_time: total_time - tracked_total_time,
     });
   }
 
   return operations;
 }
 
-export function energy_data(currentTotal){
-  const kwh = currentTotal * ENERGY_CONVERSION_UNITS['kwh'];
-  const carbon_emission_tons = kwh*ENERGY_CONVERSION_UNITS['carbon'];
-  const carbon_unit = carbon_emission_tons < 1 ? 
-      `${parseFloat(Number(carbon_emission_tons*1000).toFixed(2))} kg`: 
-      `${parseFloat(Number(carbon_emission_tons).toFixed(2))} Metric Tons`;
-  const miles = unitScale(carbon_emission_tons * ENERGY_CONVERSION_UNITS['miles'],'generic');
-  const household = homeEnergyUsedFormat(carbon_emission_tons * ENERGY_CONVERSION_UNITS['household']);
-  const phone = unitScale(carbon_emission_tons * ENERGY_CONVERSION_UNITS['phone'],'generic');
+export function energy_data(currentTotal) {
+  const kwh = currentTotal * ENERGY_CONVERSION_UNITS["kwh"];
+  const carbon_emission_tons = kwh * ENERGY_CONVERSION_UNITS["carbon"];
+  const carbon_unit =
+    carbon_emission_tons < 1
+      ? `${parseFloat(Number(carbon_emission_tons * 1000).toFixed(2))} kg`
+      : `${parseFloat(Number(carbon_emission_tons).toFixed(2))} Metric Tons`;
+  const miles = unitScale(
+    carbon_emission_tons * ENERGY_CONVERSION_UNITS["miles"],
+    "generic"
+  );
+  const household = homeEnergyUsedFormat(
+    carbon_emission_tons * ENERGY_CONVERSION_UNITS["household"]
+  );
+  const phone = unitScale(
+    carbon_emission_tons * ENERGY_CONVERSION_UNITS["phone"],
+    "generic"
+  );
 
   return {
     kwh: parseFloat(Number(kwh).toFixed(2)),
     carbon: carbon_unit,
     miles: `${miles.val} ${miles.scale}`,
     household: household,
-    phone: `${phone.val} ${phone.scale}`
-  }
-
+    phone: `${phone.val} ${phone.scale}`,
+  };
 }
 
-export function calculate_training_time(numIterations, instance){
+export function calculate_training_time(numIterations, instance) {
   // instance.x is the time for 1 iterations in msec
   // 3.6e6 to convert total training time from msec to hours, divided by the total number of GPUS
   // output is in hours
-  return numIterations * instance.x / 3.6e6 / instance.info.ngpus;
+  return (numIterations * instance.x) / 3.6e6 / instance.info.ngpus;
 }
 
-export function numberFormat(num){
+export function numberFormat(num) {
   const formatter = new Intl.NumberFormat("en-US", {
     notation: "compact",
     compactDisplay: "long",
   });
   return formatter.format(num);
 }
 
-export function currencyFormat(cost){
+export function currencyFormat(cost) {
   const scientificFormater = new Intl.NumberFormat("en-US", {
-    style: 'currency', 
-    currency: 'USD',
+    style: "currency",
+    currency: "USD",
     notation: "compact",
     compactDisplay: "short",
   });
   return scientificFormater.format(cost);
-}
\ No newline at end of file
+}
+
+export function fetchingURLErrors(type, response, validationErrors) {
```

Comment:
I think this function should function is doing too much specific work and not general work. I think you should keep the general case. For the invalid schema error, you can revert that change and have this only handle errors when fetching URLs.

Also I think you should remain the function to the active voice (e.g. getErrMsgFromInvalidURL